openg2p/models/beneficiary_to_batch_wizard.py

```diff
@@ -46,43 +46,67 @@ def create_single(self):
 
     @api.multi
     def create_batch(self):
-        beneficiaries_selected = self.env["openg2p.beneficiary"].browse(
-            self.env.context.get("active_ids")
-        )
-        program_wise = {}
+
+        self.task_create_batch(self.batch_name, self.env.context.get("active_ids"))
+        return {"type": "ir.actions.act_window_close"}
+
+    def task_create_batch(self, batch_name, beneficiary_ids):
+        beneficiaries_selected = self.env["openg2p.beneficiary"].browse(beneficiary_ids)
+        batch_ids = []
+        batch_wise = {}
+
         for b in beneficiaries_selected:
+
             if not b.bank_account_number:
-                continue
-            for program_id in b.program_ids.ids:
-                if program_id in program_wise.keys():
-                    program_wise[program_id].append(b)
-                else:
-                    program_wise[program_id] = [b]
+                raise ValidationError(
+                    "One or more beneficiaries do not have bank account details"
+                )
+
+            # if b.batch_status:
```

Comment:
Remove this if not required.

Reply:
Merging this for now. Please address this in future PR.

Reply:
Ok